lib/node_modules/@stdlib/regexp/to-json/test/test.js

```diff
@@ -0,0 +1,166 @@
+*/
+
+'use strict';
+
+// MODULES //
+
+var vm = require( 'vm' );
+var tape = require( 'tape' );
+var IS_BROWSER = require( '@stdlib/assert/is-browser' );
+var isPlainObject = require( '@stdlib/assert/is-plain-object' );
+var toJSON = require( './../lib' );
+
+
+// VARIABLES //
+
+var opts = {
+	'skip': false
+};
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof toJSON, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'if provided anything other than a regular expression, the function will throw an error', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		'5',
+		5,
+		NaN,
+		null,
+		void 0,
+		true,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), Error, 'throws when provided a ' + (typeof values[i]) );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			toJSON( value );
+		};
+	}
+});
+
+tape( 'the function returns a JSON object', function test( t ) {
+	var pattern;
+	var regex;
+	var json;
+
+	pattern = 'beep';
+	regex = new RegExp( pattern );
+	json = toJSON( regex );
+	t.strictEqual( isPlainObject( json ), true, 'returns an object' );
+	t.end();
+});
+
+tape( 'the JSON object includes a type field', function test( t ) {
+	var expected;
+	var pattern;
+	var value;
+	var json;
+
+	pattern = 'beep';
+	value = new RegExp( pattern );
+	expected = 'RegExp';
+	json = toJSON( value );
+	t.strictEqual( json.type, expected, 'type equal to ' + expected );
+	t.end();
+});
+
+tape( 'the JSON object includes a pattern field', function test( t ) {
+	var expected;
+	var values;
+	var regex;
+	var json;
+	var i;
+
+	values = [
+		'.*',
+		'ab+c',
+		'/.*^1'
+	];
+
+	expected = [
+		'.*',
+		'ab+c',
+		'\\/.*^1'
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		regex = new RegExp( values[ i ] );
+		json = toJSON( regex );
+		t.strictEqual( json.pattern, expected[ i ], 'pattern equal to \'' + expected[ i ] + '\'' );
+	}
+	t.end();
+});
+
+tape( 'the JSON object includes a flags field', function test( t ) {
+	var expected;
+	var values;
+	var regex;
+	var json;
+	var i;
+
+	values = [
+		'.*',
+		'ab+c',
+		'/.*^1'
+	];
+
+	expected = [
+		'd',
+		'dy',
+		's'
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		regex = new RegExp( values[ i ], expected[ i ] );
+		json = toJSON( regex );
+		t.strictEqual( json.flags, expected[ i ], 'flags equal to \'' + expected[ i ] + '\'' );
+	}
+	t.end();
+});
+
+opts.skip = IS_BROWSER;
+tape( 'the function supports serializing a regular expression from a different realm', opts, function test( t ) {
+	var regex;
+	var json;
+
+	regex = vm.runInNewContext( 'new RegExp( "ab+c", "g" )' );
+	json = toJSON( regex );
+
+	t.strictEqual( json.type, 'RegExp', 'type equal to regexp' );
+	t.strictEqual( json.pattern, 'ab+c', 'equal pattern' );
+	t.strictEqual( json.flags, 'g', 'equal flags' );
```

Comment:
```suggestion
	var regex;
	var json;

	regex = vm.runInNewContext( '/ab+c/g' );
	json = toJSON( regex );

	t.strictEqual( json.type, 'RegExp', 'returns expected value' );
	t.strictEqual( json.pattern, 'ab+c', 'returns expected value' );
	t.strictEqual( json.flags, 'g', 'returns expected value' );
```

In general, we use the test message "returns expected value" unless we are looping through a list of values and want to log each item in the message.

The reason for this is to mitigate copy and paste errors. Fairly common to copy test blocks within the project and using a generic message is useful to avoid devs forgetting to update and tailor the message appropriately.

In loops, logging the value is more important, as failed assertions always point to the same LOC. In this case, assertions are split over separate lines, so tracking down the failing test case is more straightforward, as one can simply use the stacktrace.

Reply:
Noted!